it/uk/gov/hmrc/minorentityidentification/repositories/JourneyDataRepositoryISpec.scala

```diff
@@ -0,0 +1,101 @@
+
+class JourneyDataRepositoryISpec extends ComponentSpecHelper with JourneyDataMongoHelper {
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    await(repo.drop)
+  }
+
+  val journeyIdKey: String = "_id"
+  val authInternalIdKey: String = "authInternalId"
+  val creationTimestampKey: String = "creationTimestamp"
+
+  "createJourney" should {
+    "successfully insert the journeyId" in {
+      await(repo.createJourney(testJourneyId, testInternalId)) mustBe testJourneyId
+    }
+  }
+  s"getJourneyData($testJourneyId)" should {
+    "successfully return all data" in {
+      insertById(testJourneyId, testInternalId, Json.obj(journeyIdKey -> testJourneyId, authInternalIdKey -> testInternalId))
+      await(repo.getJourneyData(testJourneyId, testInternalId)).map(_.-(creationTimestampKey)) mustBe
+        Some(Json.obj(journeyIdKey -> testJourneyId, authInternalIdKey -> testInternalId))
+    }
+  }
+  "updateJourneyData" should {
+    "successfully insert data" in {
+      val testKey = "testKey"
+      val testData = "test"
+      await(repo.createJourney(testJourneyId, testInternalId))
+      await(repo.updateJourneyData(testJourneyId, testInternalId, testKey, JsString(testData)))
+      await(repo.getJourneyData(testJourneyId, testInternalId)).map(json => (json \ testKey).as[String]) mustBe Some(testData)
+    }
+    "successfully update data when data is already stored against a key" in {
+      val testKey = "testKey"
+      val testData = "test"
+      val updatedData = "updated"
+      await(repo.createJourney(testJourneyId, testInternalId))
+      await(repo.updateJourneyData(testJourneyId, testInternalId, testKey, JsString(testData)))
+      await(repo.updateJourneyData(testJourneyId, testInternalId, testKey, JsString(updatedData)))
+      await(repo.getJourneyData(testJourneyId, testInternalId)).map(json => (json \ testKey).as[String]) mustBe Some(updatedData)
+    }
+
+  }
+  "removeJourneyDataField" should {
+    "successfully remove a field" in {
+      val testKey = "testKey"
+      val testData = "test"
+
+      await(repo.createJourney(testJourneyId, testInternalId))
+      await(repo.updateJourneyData(testJourneyId, testInternalId, testKey, JsString(testData)))
+      await(repo.removeJourneyDataField(testJourneyId, testInternalId, testKey))
+      await(repo.getJourneyData(testJourneyId, testInternalId)).map(_.-(creationTimestampKey)) mustBe
+        Some(Json.obj(journeyIdKey -> testJourneyId, authInternalIdKey -> testInternalId))
+    }
+    "pass successfully when the field is not present" in {
+      val testKey = "testKey"
+      val testData = "test"
+      val testSecondKey = "secondKey"
+
+      await(repo.createJourney(testJourneyId, testInternalId))
+      await(repo.updateJourneyData(testJourneyId, testInternalId, testKey, JsString(testData)))
+      await(repo.removeJourneyDataField(testJourneyId, testInternalId, testSecondKey))
```

Comment:
... it should be ...
await(repo.removeJourneyData(testJourneyId, testInternalId)) mustBe  true